xpdan/db_utils.py

```diff
@@ -0,0 +1,100 @@
+from pprint import pprint
+from .utils import _timestampstr
+
+
+def need_name_here(hdrs, key, verbose=True):
+    """Build a dictionary where the keys are the unique values for the given
+    key and the values are the positions of the headers for that key
+
+    Parameters
+    ----------
+    hdrs: list of Header objects
+        The headers to be searched
+    key: str
+        The key to be searched for
```

Comment:
The scans will be sorted by the values of this metadata key, e.g., 'sample_name'